docs/examples/robot_keyboard_2.py

```diff
@@ -3,8 +3,23 @@
 
 robot = Robot(left=(4, 14), right=(17, 18))
 
+# Get the list of available input devices
 devices = [InputDevice(device) for device in list_devices()]
-keyboard = devices[0]  # this may vary
+# Filter out everything that's not a keyboard. Keyboards are defined as any
+# device which has keys, and which specifically has keys 1..31 (roughly Esc,
+# the numeric keys, the first row of QWERTY plus a few more) and which does
+# *not* have key 0 (reserved)
+must_have = {i for i in range(1, 32)}
+must_not_have = {0}
+devices = [
+    device
+    for device in devices
+    for keys in (set(device.capabilities().get(ecodes.EV_KEY, [])),)
+    if must_have.issubset(keys)
+    and must_not_have.isdisjoint(keys)
+]
```

Comment:
Hmm, is it the similarity of the words that's the issue? Would "dev" be better to full distinguish the result variable from the source variable "devices"?